ts/receiver/mediumGroups.ts

```diff
@@ -342,73 +342,23 @@ async function handleMediumGroupChange(
     convo.set('isKickedFromGroup', true);
     // Disable typing:
     convo.updateTextInputState();
+    window.SwarmPolling.removePubkey(groupId);
   }
 
   await convo.commit();
 
-  await removeFromCache(envelope);
-}
-
-async function handleQuit(
-  envelope: EnvelopePlus,
-  groupUpdate: SignalService.MediumGroupUpdate
-) {
-  const quitter = envelope.source;
-  const groupId = toHex(groupUpdate.groupPublicKey);
-
-  const quitterPrimary = await MultiDeviceProtocol.getPrimaryDevice(quitter);
-
-  const maybeConvo = await window.ConversationController.get(groupId);
-
-  if (!maybeConvo) {
-    window.log.warn('Received QUIT for a non-existing medium group');
-    await removeFromCache(envelope);
-    return;
-  }
-
-  const convo = maybeConvo;
-
-  // 1. Remove primary device from members:
-
-  const members = convo.get('members');
-
-  const membersUpdated = _.without(members, quitterPrimary.key);
-
-  convo.set({ members: membersUpdated });
-
-  convo.commit();
-
-  // 2. Show message (device left the group);
-
-  await SenderKeyAPI.addUpdateMessage(
-    convo,
-    { leavingMembers: [quitterPrimary.key] },
-    'incoming'
-  );
-
-  const ourNumber = (await UserUtil.getCurrentDevicePubKey()) as string;
-  const primary = await UserUtil.getPrimary();
-
-  if (quitterPrimary.key === primary.key) {
-    convo.set('isKickedFromGroup', true);
-    // Disable typing:
-    convo.updateTextInputState();
-    await convo.commit();
+  if (diff.leavingMembers && diff.leavingMembers.length > 0) {
+    // Send out the user's new ratchet to all members (minus the removed ones) using established channels
```

Comment:
Didn't we decide that whoever kicked a member will re-generate everyone's keys instead?

Reply:
I matched what was done on android and after speaking with Niels he told me this is the way we do it now

Reply:
This was for a few different reasons. From what I can recall now:

• Otherwise you could pretend to generate new ratchets but actually just redistribute the old ones (allowing the removed member to eavesdrop)
• This was you can re-use this code for group leaving (i.e. removing yourself)

Admittedly it's a while ago so I'm not 100% sure anymore what the reasoning was at the time